cob_sick_s300/common/include/cob_sick_s300/TelegramS300.h

```diff
@@ -229,47 +255,87 @@ class TelegramParser {
 		ntoh(tc1_);
 		if(debug) print(tc1_);
 
-		if(tc1_.size*2+JUNK_SIZE>(int)max_size) {
-			if(debug) std::cout<<"invalid header size"<<std::endl;
-			return false;
-		}
-
 		tc2_ = *((TELEGRAM_COMMON2*)(buffer+sizeof(TELEGRAM_COMMON1)));
 		tc3_ = *((TELEGRAM_COMMON3*)(buffer+(sizeof(TELEGRAM_COMMON1)+sizeof(TELEGRAM_COMMON2))));
 
 		TELEGRAM_TAIL tt;
 		uint16_t crc;
-
+		int full_data_size = 0;
+
+		// The size reported by the protocol varies depending on the calculation which is different depending
+		// on several factors.
+		// The calculation is described on pp. 70-73 in:
+		// https://www.sick.com/media/dox/1/91/891/Telegram_listing_S3000_Expert_Anti_Collision_S300_Expert_de_en_IM0022891.PDF
+		//
+		// Also, the size is reported as 16bit-words = 2 bytes...
+		//
+		// For the old protocol/compatability mode:
+		// "The telegram size is calculated ... starting with the ... 5. byte ... up to and including the ... CRC."
 		if(tc2_.protocol_version==0x102)
```

Comment:
Detecting if the protocol is not supported could IMO only be done via checking the protocl number. However, this node could use some more error handling in this parser :-)

However, this would then also be another issue...